src/App/Pages/Vault/ScanPage.xaml.cs

```diff
@@ -50,66 +49,98 @@ public ScanPage(Action<string> callback)
+                        while (!autofocusCts.IsCancellationRequested)
                         {
-                            if (!autofocusCts.IsCancellationRequested)
+                            await Task.Delay(TimeSpan.FromSeconds(2), autofocusCts.Token);
+                            await Device.InvokeOnMainThreadAsync(() =>
                             {
-                                try
-                                {
-                                    _zxing.AutoFocus();
-                                }
-                                catch (Exception ex)
+                                if (!autofocusCts.IsCancellationRequested)
                                 {
-                                    _logger.Value.Exception(ex);
+                                    try
+                                    {
+                                        _zxing.AutoFocus();
+                                    }
+                                    catch (Exception ex)
+                                    {
+                                        _logger.Value.Exception(ex);
+                                    }
                                 }
-                            }
-                        });
+                            });
+                        }
                     }
-                }
-                catch (TaskCanceledException) { }
-                catch (Exception ex)
-                {
-                    _logger.Value.Exception(ex);
-                }
-            }, autofocusCts.Token);
-            _pageIsActive = true;
-            AnimationLoopAsync();
+                    catch (TaskCanceledException) { }
+                    catch (Exception ex)
+                    {
+                        _logger.Value.Exception(ex);
+                    }
+                }, autofocusCts.Token);
+                _pageIsActive = true;
+                AnimationLoopAsync();
+            }
         }
 
         protected override async void OnDisappearing()
         {
-            _autofocusCts?.Cancel();
-            if (_continuousAutofocusTask != null)
+            if (_zxing != null)
             {
-                await _continuousAutofocusTask;
+                _autofocusCts?.Cancel();
+                if (_continuousAutofocusTask != null)
+                {
+                    await _continuousAutofocusTask;
+                }
+                _zxing.IsScanning = false;
+                _pageIsActive = false;
             }
-            _zxing.IsScanning = false;
-            _pageIsActive = false;
             base.OnDisappearing();
         }
 
+        private void InitScanner()
+        {
+            try
+            {
+                if (ViewModel.HasCameraPermission && ViewModel.ShowScanner && _zxing == null)
+                {
+                    _zxing = new ZXingScannerView();
+                    _zxing.OnScanResult -= OnScanResult;
+                    _zxing.OnScanResult += OnScanResult;
```

Comment:
Here you are initializing a new `ZXingScannerView` so `_zxing.OnScanResult -= OnScanResult;` does nothing, given that it's a fresh new instance. Thus you're subscribing to `OnScanResult` but you never unsubscribe from that, thus you might have a leak here.